From: Hiroaki Nakamura <hnakamur@gmail.com>
Date: Tue, 19 Jun 2018 14:25:40 +0900
Subject: Fix checking -lcrypto for OpenSSL 1.1

---
 configure    | 44 +++++++++++++++++++++++++++++++++++++++++++-
 configure.ac |  8 ++++++--
 2 files changed, 49 insertions(+), 3 deletions(-)

diff --git a/configure b/configure
index d51b981..d68d9b8 100755
--- a/configure
+++ b/configure
@@ -5231,7 +5231,46 @@ if eval test \"x\$"$as_ac_Lib"\" = x"yes"; then :
 
 $as_echo "#define HAVE_SSL 1" >>confdefs.h
 
-					{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for OPENSSL_add_all_algorithms_noconf in -lcrypto" >&5
+					{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for OPENSSL_init_crypto in -lcrypto" >&5
+$as_echo_n "checking for OPENSSL_init_crypto in -lcrypto... " >&6; }
+if ${ac_cv_lib_crypto_OPENSSL_init_crypto+:} false; then :
+  $as_echo_n "(cached) " >&6
+else
+  ac_check_lib_save_LIBS=$LIBS
+LIBS="-lcrypto  $LIBS"
+cat confdefs.h - <<_ACEOF >conftest.$ac_ext
+/* end confdefs.h.  */
+
+/* Override any GCC internal prototype to avoid an error.
+   Use char because int might match the return type of a GCC
+   builtin and then its argument prototype would still apply.  */
+#ifdef __cplusplus
+extern "C"
+#endif
+char OPENSSL_init_crypto ();
+int
+main ()
+{
+return OPENSSL_init_crypto ();
+  ;
+  return 0;
+}
+_ACEOF
+if ac_fn_c_try_link "$LINENO"; then :
+  ac_cv_lib_crypto_OPENSSL_init_crypto=yes
+else
+  ac_cv_lib_crypto_OPENSSL_init_crypto=no
+fi
+rm -f core conftest.err conftest.$ac_objext \
+    conftest$ac_exeext conftest.$ac_ext
+LIBS=$ac_check_lib_save_LIBS
+fi
+{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_crypto_OPENSSL_init_crypto" >&5
+$as_echo "$ac_cv_lib_crypto_OPENSSL_init_crypto" >&6; }
+if test "x$ac_cv_lib_crypto_OPENSSL_init_crypto" = xyes; then :
+  LIBSSL="${LIBSSL} -lcrypto"
+else
+  { $as_echo "$as_me:${as_lineno-$LINENO}: checking for OPENSSL_add_all_algorithms_noconf in -lcrypto" >&5
 $as_echo_n "checking for OPENSSL_add_all_algorithms_noconf in -lcrypto... " >&6; }
 if ${ac_cv_lib_crypto_OPENSSL_add_all_algorithms_noconf+:} false; then :
   $as_echo_n "(cached) " >&6
@@ -5269,6 +5308,9 @@ fi
 $as_echo "$ac_cv_lib_crypto_OPENSSL_add_all_algorithms_noconf" >&6; }
 if test "x$ac_cv_lib_crypto_OPENSSL_add_all_algorithms_noconf" = xyes; then :
   LIBSSL="${LIBSSL} -lcrypto"
+fi
+
+
 fi
 
 					LIBSSL="${LIBSSL}"
diff --git a/configure.ac b/configure.ac
index b02dd7c..c3c172e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -122,9 +122,13 @@ AS_IF([test "x$with_ssl" != xno],
 					[LIBSSL="-lssl"
 					AC_DEFINE([HAVE_SSL], [1], [Define if you have ssl])
 					AC_CHECK_LIB([crypto],
-								 [OPENSSL_add_all_algorithms_noconf],
+								 [OPENSSL_init_crypto],
 								 [LIBSSL="${LIBSSL} -lcrypto"],
-								 [])
+								 [AC_CHECK_LIB([crypto],
+								 	[OPENSSL_add_all_algorithms_noconf],
+								 	[LIBSSL="${LIBSSL} -lcrypto"],
+									[])
+								 ])
 					AC_SUBST([LIBSSL], ["${LIBSSL}"])
 					],
 					[if test "x$with_ssl" != xcheck; then
